BitmarkRegistry/Services/SyncStorage/TransactionStorage.swift

```diff
@@ -8,62 +8,85 @@
-    let (txs, assets, blocks) = try TransactionService.listAllTransactions(ownerNumber: owner.getAccountNumber(), at: latestOffset, direction: .later)
-    guard !txs.isEmpty else { return nil }
+  func syncData(for bitmarkR: BitmarkR) throws {
+    let realm = try ownerRealm()
+    var latestOffset: Int64 = 0
 
-    var txsWithRelative = TransactionsWithRelative(assets: assets, blocks: blocks, txs: txs)
+    repeat {
+      let (txs, blocks) = try TransactionService.listAllTransactions(of: bitmarkR.id, at: latestOffset, direction: .later)
+      guard !txs.isEmpty else { return }
 
-    if notifyNew {
-      Global.addAssets(txsWithRelative.assets)
-      Global.addBlocks(txsWithRelative.blocks)
-      DispatchQueue.main.async { [weak self] in
-        self?.delegate?.receiveNewRecords(txsWithRelative.txs)
-      }
-    }
+      try storeData(in: realm, txs: txs, relations: (assets: nil, blocks: blocks))
+      latestOffset = txs.last!.offset
+      if txs.count < 100 { break }
+    } while true
+  }
+
+  override func syncData() throws {
+    let backgroundOwnerRealm = try ownerRealm()
+    var latestOffset = getLatestOffsetR(in: backgroundOwnerRealm)?.offset ?? 0
 
-    let baseOffset = txs.last!.offset // the last offset is the latest offset cause response transactions is asc-offset
-    let txsURL = self.fileURL(pathName: baseOffset)
+    repeat {
+      let (txs, assets, blocks) = try TransactionService.listAllTransactions(ownerNumber: owner.getAccountNumber(), at: latestOffset, direction: .later)
+      guard !txs.isEmpty else { return }
 
-    if latestOffset == 0 {
-      try txsWithRelative.store(in: txsURL)
-    } else {
-      try mergeNewTxs(txsURL, txsWithRelative)
+      try storeData(in: backgroundOwnerRealm, txs: txs, relations: (assets: assets, blocks: blocks))
+      latestOffset = txs.last!.offset
+      if txs.count < 100 { break }
+    } while true
+
+    try backgroundOwnerRealm.write {
+      backgroundOwnerRealm.add(LatestOffsetR(value: ["Transaction", latestOffset]), update: .modified)
     }
-    return baseOffset
   }
 
-  // Merge new bitmarks into the bitmarks file
-  fileprivate func mergeNewTxs(_ newTxsURL: URL, _ txsWithAsset: TransactionsWithRelative) throws {
-    guard let latestPathName = Global.latestOffset["Transaction"] else { return }
-    let latestTxsURL = fileURL(pathName: latestPathName)
-    var oldTxsWithAsset = try TransactionsWithRelative(from: latestTxsURL)
-    try oldTxsWithAsset.merge(
-      with: txsWithAsset,
-      from: latestTxsURL,
-      to: newTxsURL
-    )
+  fileprivate func storeData(in realm: Realm, txs: [Transaction], relations: (assets: [Asset]?, blocks: [Block])) throws {
+    try txs.forEach { (tx) in
+      let txR = TransactionR(tx: tx)
+      try realm.write {
+        if let assets = relations.assets, let asset = assets.first(where: { $0.id == tx.asset_id }) {
+          let assetR = AssetR(asset: asset)
+          realm.add(assetR, update: .modified)
```

Comment:
Once an asset is on blockchain, it will never get modified. Modifying here might cause an expensive process and introduce a notification to our views.